components/time.rst

```diff
@@ -10,7 +10,184 @@ Time
 
 The ``time`` component allows you to set up real time clock time sources for ESPHome.
 You can then get the current time in :ref:`lambdas <config-lambda>`.
-Currently only SNTP (internet-based), Home Assistant and GPS time sources are supported.
+
+.. _base_time_config:
+
+Base Time Configuration
+-----------------------
+
+All time configuration schemas inherit these options.
+
+Configuration variables:
+************************
+
+- **id** (*Optional*, :ref:`config-id`): Specify the ID of the time for use in lambdas.
+- **timezone** (*Optional*, string): Manually tell ESPHome what time zone to use with `this format
+  <https://www.gnu.org/software/libc/manual/html_node/TZ-Variable.html>`__ (warning: the format is quite complicated)
+  or the simpler `TZ database name <https://en.wikipedia.org/wiki/List_of_tz_database_time_zones>`__ in the form
+  <Region>/<City>. ESPHome tries to automatically infer the time zone string based on the time zone of the computer
+  that is running ESPHome, but this might not always be accurate.
+- **on_time** (*Optional*, :ref:`Automation <automation>`): Automation to run at specific intervals using
+  a cron-like syntax. See :ref:`time-on_time`.
+- **on_time_sync** (*Optional*, :ref:`Automation <automation>`): Automation to run when the time source
```

Comment:
Overall PR is ok, but we can beware of terminology better:

Sync is already used, mostly in code, to indicate the ESP clock is updated with some time source, which is what's happening here. I don't understand this short description, wouldn't something like: 

Automation to run after the time source synchronizes the internal clock. See ...

Reply:
How about `on_time_update`? - That could certainly also be missunderstood because the time is essentially updated every second... The same would apply to `on_time_change`. Maybe `on_time_set`?
Im very open to suggestions!

Reply:
It's ok on sync. I was thinking out loud. Ds1370 has read and write but those are actions. 
It's a read action call causing the sync event trigger which is odd but May be ok, a sync action will be confusing. 
Btw ds1370 can store data? Wouldn't read write be confusing then?
As 1.16 is in beta we can change that before making it a breaking change. Not a request but a comment

Reply:
You are right: DS1307 has a small (56 bytes) NVRAM section to store data. My idea was to read all data incl. time an NVRAM in one call but it is probably really better to have separate calls to be able to update the data section without an update to the time registers. I'll change that later today.

My proposal for the actions would be  `read_time` and `write_time` to make it clear the these actions only store or retrieve the time information. I'll add an interface for the general purpose registers later.

A `read_time` will indeed trigger an `on_time_sync` event which is intentional: other components can register to be updated whenever the RTC is used to update the system clock.
Optionally, I am still thinking of another wording for the `on_time_sync` event. I think `sync` is consistent, because the event shall be triggered whenever `synchronize_epoch_()`  is called to update the system time.
I'd also be 100% OK to rename it to something like `on_time_set`. Please post your opinions or other options here. Otherwise I'll leve it as it is. ;-)

Reply:
`on_time_sync` ok